Api test + Allure

PavAnufriev opened the pull request:
Api test + Allure
Summary
Добавлены API-тесты для Stellar Burgers с использованием POJO.
Подключены REST Assured, JUnit 4 и Allure.
Настроен Surefire для запуска и генерации Allure-результатов.
Что сделано по заданию
Создание пользователя:
уникальный пользователь — позитивный сценарий
повторная регистрация — 403 User already exists
отсутствие обязательного поля — 403 Email, password and name are required fields
Логин пользователя:
успешный вход
неверные логин/пароль — 401 email or password are incorrect
Создание заказа:
с авторизацией и ингредиентами — 200
без авторизации — допустим 200/401/403 (в зависимости от бэкенда)
без ингредиентов — 400 Ingredient ids must be provided
с невалидным хешем — 500 или 400 (учтены оба варианта)
Техстек
Maven, JUnit 4, REST Assured, Allure
POJO-модели: User, LoginRequest, OrderRequest (+ простые респонсы при необходимости)
Базовая спецификация RestAssured + Allure фильтр

PavAnufriev committed:
Api test + Allure

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/OrderTests.java:
```diff
@@ -0,0 +1,96 @@
+    }
+
+    @Test
+    @Story("Create order with auth and ingredients")
```
⚠️Можно улучшить. В allure есть еще одна аннотация для описания теста DisplayName. Добавь ее, пожалуйста, во всех тестах проекта

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/OrderTests.java:
```diff
@@ -0,0 +1,96 @@
+    @Description("Создание заказа: с авторизацией и с ингредиентами")
+    public void createOrder_withAuth_withIngredients_success() {
+        // Register user and get token
+        User user = User.random();
```
⛔️Нужно исправить. Шаги создания пользователя   и получения ингредиентов нужно вынести в метод с аннотацией before

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/OrderTests.java:
```diff
@@ -0,0 +1,96 @@
+
+        // Create order
+        orderSteps.createOrder(accessToken, new OrderRequest(ingBody))
+                .statusCode(200)
```
⚠️Можно улучшить. Вместо числовых кодов ответов, лучше использовать коды из библиотеки static org.apache.http.HttpStatus, так как они там строковые и несут смысл, что повышает читаемость и не надо вспоминать, что есть что. Пример кода, statusCode(SC_CREATED).

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/OrderTests.java:
```diff
@@ -0,0 +1,96 @@
+        List<String> ingBody = ingredients.subList(0, Math.min(2, ingredients.size()));
+
+        orderSteps.createOrderWithoutAuth(new OrderRequest(ingBody))
+                .statusCode(anyOf(is(200), is(401), is(403))); // backend may allow or deny; check code presence only
```
⛔️Нужно исправить. Укажи , пожалуйста, только один код

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/OrderTests.java:
```diff
@@ -0,0 +1,96 @@
+    public void createOrder_wrongIngredientHash_internalServerError() {
+        // Try both with and without auth
+        orderSteps.createOrderWithoutAuth(new OrderRequest(List.of("invalid_hash")))
+                .statusCode(anyOf(is(500), is(400))); // backend can return 500 per spec, sometimes 400 in practice
```
⛔️Нужно исправить. Укажи , пожалуйста, только один код

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/OrderTests.java:
```diff
@@ -0,0 +1,96 @@
+    @Test
+    @Story("Create order with wrong ingredient hash")
+    @Description("Создание заказа: с неверным хешем ингредиентов")
+    public void createOrder_wrongIngredientHash_internalServerError() {
```
⛔️Нужно исправить. Неверный нейминг. Мы не используем спецсимволы в названии методов

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/UserTests.java:
```diff
@@ -0,0 +1,99 @@
+    @Story("Register user missing required field")
+    @Description("Api test + Allure")
+    public void registerUser_missingField_forbidden() {
+        User user = new User(null, "Passw0rd!", "NoEmail");
```
⛔️Нужно исправить. Должно быть минимум три теста. Необходимо проверить обязательность каждого поля

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/UserTests.java:
```diff
@@ -0,0 +1,99 @@
+                .body("user.email", equalTo(user.email))
+                .body("accessToken", notNullValue());
+
+        accessToken = resp.extract().path("accessToken");
```
⛔️Нужно исправить Получение токена нужно вынести в метод с аннотацией after. Шаг должен выполняться даже после падения теста

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/UserTests.java:
```diff
@@ -0,0 +1,99 @@
+    }
+
+    @Test
+    @Story("Login existing user")
```
⛔️Нужно исправить Тесты на логин нужно вынести в отдельный класс. Для каждой ручки свои тесты

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/UserTests.java:
```diff
@@ -0,0 +1,99 @@
+    @Test
+    @Story("Login with wrong credentials")
+    @Description("Api test + Allure")
+    public void loginWrongCredentials_unauthorized() {
```
⛔️Нужно исправить. Должно быть два теста 
1. Неверный логин
2. Неверный пароль

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/OrderTests.java:
```diff
@@ -0,0 +1,96 @@
+                .body("order.number", notNullValue())
+                .body("name", notNullValue());
+    }
+
```
⛔️Нужно исправить.  Нет тестов на получение списка заказов

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/models/User.java:
```diff
@@ -0,0 +1,20 @@
+    }
+
+    public static User random() {
+        String suffix = String.valueOf(System.currentTimeMillis());
```
⚠️Можно улучшить. Лучше использовать готовые решения JavaFaker/DataFaker, так как это сокращает нашу работу.

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/steps/UserSteps.java:
```diff
@@ -0,0 +1,34 @@
+        return given()
+                .body(body)
+                .when()
+                .post("/api/auth/register")
```
⚠️Можно улучшить. Константы с указанием пути можно вынести в отдельный класс, так с ними будет удобнее работать. Назови его, например, Endpoints

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/steps/UserSteps.java:
```diff
@@ -0,0 +1,34 @@
+                .when()
+                .delete("/api/auth/user")
+                .then();
+    }
```
⛔️Нужно исправить.  Приложи, пожалуйста, аллюр отчет

okryukova replied:
Нет отчета. Папка allure-results

okryukova replied:
Не исправлено

okryukova commented on pom.xml:
```diff
@@ -34,6 +71,29 @@
+            </plugin>
+
             <!-- JaCoCo plugin -->
             <plugin>
```
⛔️Нужно исправить.  Для данного задания плагин JaCoCo не нужен

PavAnufriev committed:
Api tests + Allure: split login tests, robust cleanup, constants for endpoints, adjust status expectations; remove JaCoCo

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/OrderTests.java:
```diff
@@ -0,0 +1,126 @@
+    @Description("Создание заказа с неверным хешем ингредиентов")
+    public void createOrderWrongIngredientHashBadRequest() {
+        orderSteps.createOrderWithoutAuth(new OrderRequest(List.of("invalid_hash")))
+                .statusCode(SC_BAD_REQUEST); // API returns 400 for invalid ingredient hash
```
⛔️Нужно исправить.  Должен быть код 500

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/OrderTests.java:
```diff
@@ -0,0 +1,126 @@
+                .statusCode(SC_BAD_REQUEST); // API returns 400 for invalid ingredient hash
+    }
+
+    @Test
```
⛔️Нужно исправить. Для каждой ручки нужно создать отдельный тестовый класс

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/UserLoginTests.java:
```diff
@@ -0,0 +1,76 @@
+    @Description("Вход под существующим пользователем")
+    public void loginExistingUserSuccess() {
+        User user = User.random();
+        userSteps.register(user).statusCode(SC_OK);
```
⛔️Нужно исправить.  Шаг создания юзера нужно вынести в метод с аннотацией before

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/UserLoginTests.java:
```diff
@@ -0,0 +1,76 @@
+                .body("user.email", equalTo(user.email))
+                .body("accessToken", notNullValue());
+
+        accessToken = resp.extract().path("accessToken");
```
⛔️Нужно исправить.  Получение токена в метод с аннотацией after. Шаг должен выполняться даже если тест упал

okryukova replied:
Не исправлено

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/UserRegistrationTests.java:
```diff
@@ -0,0 +1,109 @@
+                .body("user.name", equalTo(user.name))
+                .body("accessToken", notNullValue());
+
+        accessToken = resp.extract().path("accessToken");
```
⛔️Нужно исправить. Получение токена в метод с аннотацией after. Шаг должен выполняться даже если тест упал

okryukova replied:
не исправлено

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/steps/OrderSteps.java:
```diff
@@ -0,0 +1,56 @@
+                .then();
+    }
+
+    public ValidatableResponse createOrderWithoutAuth(OrderRequest body) {
```
⛔️Нужно исправить. Для всех методов, описывающих шаги теста, необходимо использовать аннотацию @Step

PavAnufriev committed:
Рефакторинг тестов и добавление интеграции с Allure

- Разделение OrderTests на OrderCreationTests и OrderRetrievalTests
- Добавление нового класса IngredientsTests
- Добавление аннотаций Allure (@Feature, @Story, @Description, @Step)
- Обновление pom.xml с зависимостями и плагинами Allure
- Добавление падающего теста для получения заказов незалогиненным пользователем
- Добавление GitHub Actions workflow для отчетов Allure
- Добавление документации и README

okryukova commented on src/test/java/ru/yandex/praktikum/stellarburgers/UserLoginTests.java:
```diff
@@ -0,0 +1,86 @@
+                .body("message", equalTo("email or password are incorrect"));
+
+        // Получаем правильный токен для cleanup
+        accessToken = userSteps.login(new LoginRequest(testUser.email, testUser.password))
```
⛔️Нужно исправить. Получение токена в метод с аннотацией after. Шаг должен выполняться даже если тест упал

PavAnufriev committed:
Исправление замечаний ревьюера

- Исправлена логика получения токенов: теперь все токены для cleanup получаются в @After методах
- Добавлена папка allure-results/ с исходными результатами тестов для демонстрации ревьюеру
- Добавлена папка allure-report/ с готовым HTML отчетом для просмотра
- Создана инструкция ALLURE_GUIDE.md с подробным описанием работы с отчетами
- Временно убраны из .gitignore папки с результатами тестов для демонстрации
